ui/apps/platform/src/Containers/Search/SearchResults.tsx

```diff
@@ -138,16 +132,111 @@ function SearchResults({
     }
 
     function onTabClick(_event, eventKey) {
-        const selectedTab = defaultTabs[eventKey];
-        setGlobalSearchCategory(selectedTab.category);
+        setSearchCategory(eventKey);
+    }
+
+    if (searchOptions.length === 0) {
+        return (
+            <Bullseye>
+                <EmptyStateTemplate title="Search all data" headingLevel="h1">
+                    Choose one or more filter values to search.
+                </EmptyStateTemplate>
+            </Bullseye>
+        );
     }
 
-    const contents = sortedRows.length ? (
+    /*
+     * Replace searchCounts.reduce(…) with searchResults.length after future improvement:
+     * replace redundant requests for each selected tab categories
+     * with filtering of the response for all categories
+     */
+    function getTabCategoryCount(tabCategory: TabCategory) {
+        return tabCategory === 'SEARCH_UNSET'
+            ? searchCounts.reduce((total, { count }) => total + Number(count), 0)
+            : searchCounts.find(({ category }) => category === tabCategory)?.count ?? 0;
+    }
+
+    /* eslint-disable no-nested-ternary */
+    return (
+        <div className="bg-base-100 flex-1" data-testid="global-search-results">
+            <Title headingLevel="h1" className="px-4 py-4">
+                Search
+            </Title>
+            <section className="h-full">
+                <Tabs activeKey={searchCategory} onSelect={onTabClick}>
+                    {tabs.map(({ tabCategory, text }) => (
+                        <Tab
+                            key={tabCategory}
+                            eventKey={tabCategory}
+                            title={
+                                <TabTitleText>
+                                    <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+                                        <FlexItem>{text}</FlexItem>
+                                        <FlexItem>
+                                            <Badge isRead>{getTabCategoryCount(tabCategory)}</Badge>
+                                        </FlexItem>
+                                    </Flex>
+                                </TabTitleText>
+                            }
+                        />
+                    ))}
+                </Tabs>
+                {tabs.map(({ tabCategory, text }) => (
+                    <TabContent
+                        eventKey={tabCategory}
+                        className="overflow-auto"
+                        id={tabCategory}
+                        aria-label={text}
+                        key={tabCategory}
+                        hidden={tabCategory !== searchCategory}
+                    >
+                        {tabCategory !== searchCategory ? null : sortedRows.length === 0 ? (
```

Comment:
Controversial nested ternary is probably better than incorrect hidden markup (for testing, maintainability, and possibly accessibility.)

 I have high hopes that pattern matching will come to JS and replace nested ternaries and switch statements before I retire though. 😄